test/cases/testdata/objectdig/test-objectdig.yaml

```diff
@@ -0,0 +1,127 @@
+cases:
```

Comment:
Do we ever need to escape the key? Does ["a.b.c"] work as a path if there's a key with that name? Or ["a\"bc"], etc. At least the former isn't too uncommon to see, so would be good to have a test case. Similarly for unicode in keys.

The current tests are great though!

Reply:
Thanks Anders 😊 . I haven't been able to think of an cases were we'd need to escape keys. Perhaps not all that clear, but I had added that as a test case here: https://github.com/open-policy-agent/opa/pull/4166/files#diff-6c6841e8969caae7396094644b6f34fa3728f0ae74f4520134eb64ba356abed7R79-R88 where there's an array as one of the keys in the path. Seems to work ☺️ 

I suppose conceptually I think of this as an ordered list of terms, where a term can be anything - even other arrays.